lib/arisaid/usergroups.rb

```diff
@@ -47,26 +47,65 @@ def apply
 
       local.each do |src|
         dst = remote.find_by(name: src['name'])
-        case
-        when dst.nil? then create src
-        when same?(src, dst) then nil
-        when changed?(src, dst) then update(src)
-        else
+
+        if dst.nil?
+          puts "create usergroup: #{src['name']}"
+          puts "  + description: #{src['description']}"
+          src['users'].flatten.each do |user|
+            puts "  + user #{user}"
+          end
+
+          create(src) unless Arisaid.read_only?
+          next
+        end
+
+        next if same?(src, dst)
+
+        if changed?(src, dst)
+          puts "update usergroup: #{src['name']}"
+          update(src)
+        end
+
+        if users_changed?(src, dst)
+          add = src['users'].flatten.sort  - dst['users'].flatten.sort
+          delete = dst['users'].flatten.sort - src['users'].flatten.sort
+          add.each do |u|
+            puts "  + user #{u}"
+          end
+          delete.each do |u|
+            puts "  - user #{u}"
+          end
```

Comment:
I always seem to see diff. Correct?

Reply:
This should to display only dry-run, I fixed.